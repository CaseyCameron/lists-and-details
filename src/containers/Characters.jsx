import React, { useEffect, useState } from 'react';
import CharacterList from '../components/characters/CharacterList';
import { fetchCharacters } from '../services/CharacterFetch';

const ListAvatarCharacters = () => {
  const [loading, setLoading] = useState(true);
  const [characters, setCharacters] = useState([]);

  useEffect(async () => {
    const characters = await fetchCharacters();
    setCharacters(characters);
    setLoading(false);
  }, []);

  if (loading) return <h1>Loading...</h1>;
  return <CharacterList characters={characters} />;
};

export default ListAvatarCharacters;