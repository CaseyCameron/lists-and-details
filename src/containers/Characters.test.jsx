import React from 'react';
import { render, screen } from '@testing-library/react';
import Characters from './Characters';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Avatar Characters Container', () => {
  it('Displays a list of Avatar characters', async () => {
    render(<Router><Characters /></Router>);

    screen.getByText('Loading...');

    const photoUrls = await screen.findAllByRole('img');
    expect(photoUrls).toHaveLength(20);
    expect(photoUrls).toMatchSnapshot();
  });
});