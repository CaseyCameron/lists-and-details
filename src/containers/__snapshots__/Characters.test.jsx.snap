// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Characters Container Displays a list of Avatar characters 1`] = `
Array [
  <img
    alt="Analay"
    src="https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201"
  />,
  <img
    alt="Arik"
    src="https://vignette.wikia.nocookie.net/avatar/images/6/6f/Arik.png/revision/latest?cb=20140808184002"
  />,
  <img
    alt="Baatar Jr."
    src="https://vignette.wikia.nocookie.net/avatar/images/4/45/Baatar_Jr..png/revision/latest?cb=20150409193510"
  />,
  <img
    alt="Bolin"
    src="https://vignette.wikia.nocookie.net/avatar/images/d/d4/Bolin.png/revision/latest?cb=20150406232314"
  />,
  <img
    alt="Chan (Fire Nation admiral)"
  />,
  <img
    alt="Chit Sang"
    src="https://vignette.wikia.nocookie.net/avatar/images/8/81/Chit_Sang.png/revision/latest?cb=20121021000707"
  />,
  <img
    alt="Asami Sato"
    src="https://vignette.wikia.nocookie.net/avatar/images/b/b6/Asami_Sato.png/revision/latest?cb=20150407110150"
  />,
  <img
    alt="Bum-Ju"
    src="https://vignette.wikia.nocookie.net/avatar/images/b/b6/Bum-Ju.png/revision/latest?cb=20140730103156"
  />,
  <img
    alt="Butler"
    src="https://vignette.wikia.nocookie.net/avatar/images/8/82/Butler.png/revision/latest?cb=20121107115044"
  />,
  <img
    alt="Broadsword man"
    src="https://vignette.wikia.nocookie.net/avatar/images/9/98/Broadsword_man.png/revision/latest?cb=20130630171553"
  />,
  <img
    alt="Aiwei"
    src="https://vignette.wikia.nocookie.net/avatar/images/0/07/Aiwei_reveals_Korra%27s_lie.png/revision/latest?cb=20140808130551"
  />,
  <img
    alt="Animal control officer"
    src="https://vignette.wikia.nocookie.net/avatar/images/5/58/Animal_control_officer.png/revision/latest?cb=20140504160655"
  />,
  <img
    alt="Chow"
    src="https://vignette.wikia.nocookie.net/avatar/images/5/51/Chow.png/revision/latest?cb=20140730173827"
  />,
  <img
    alt="Circus master"
    src="https://vignette.wikia.nocookie.net/avatar/images/a/a5/Circus_master.png/revision/latest?cb=20130706153819"
  />,
  <img
    alt="Bujing"
    src="https://vignette.wikia.nocookie.net/avatar/images/7/76/Bujing.png/revision/latest?cb=20130714152817"
  />,
  <img
    alt="Buttercup Raiko"
    src="https://vignette.wikia.nocookie.net/avatar/images/e/ed/Buttercup_Raiko.png/revision/latest?cb=20150409193510"
  />,
  <img
    alt="Cabbage merchant"
    src="https://vignette.wikia.nocookie.net/avatar/images/2/2f/Cabbage_merchant.png/revision/latest?cb=20140112200908"
  />,
  <img
    alt="Calm man"
    src="https://vignette.wikia.nocookie.net/avatar/images/b/b6/Calm_man.png/revision/latest?cb=20130814093330"
  />,
  <img
    alt="Chin"
    src="https://vignette.wikia.nocookie.net/avatar/images/6/6f/Chin_the_Great.png/revision/latest?cb=20140215111604"
  />,
  <img
    alt="Chit Sang's friend"
    src="https://vignette.wikia.nocookie.net/avatar/images/6/6e/Chit_Sang%27s_friend.png/revision/latest?cb=20121120131829"
  />,
]
`;
